### DISCUSSION TITLE
Check is runner is idle?

### DISCUSSION THREAD

**Discussion by igorwojda** [community/community#160563] [Actions]
I am experiencing an issue with self-hosted runners getting stuck https://github.com/actions/runner/issues/2532 and I am looking for a workaround.
I was thinking about restarting the runner services once in a while. Is there a way to check is runner is idle (does not run any job at the moment) so I could safely restart it? or perhaps there is a way to restart a runner after the job is completed?

**Comment by DenisPalnitsky** [comment]
You probably already figured out the answer but in case others will stumble upon it:
1. To find out if the runner is Idle you can use List runners Rest API  https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#list-self-hosted-runners-for-an-organization
2. To restart after job finishes you can configure [ACTIONS_RUNNER_HOOK_JOB_COMPLETED](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job#triggering-the-scripts) to run a `shutdown -r` or `ec2 reboot-instances` if you are on aws
